zeroconf.py

```diff
@@ -1082,7 +1082,11 @@ def __init__(self, type, name, address=None, port=None, weight=0,
                     bytes for the text field)
         server: fully qualified name for service host (defaults to name)"""
 
-        if not name.endswith(type):
+        compare_type = type
+        if '_sub' in type:
```

Comment:
This test should likely be for '_sub.'  This test seems narrow enough that it could match legal service type names.